trunner/tools/gdb.py

```diff
@@ -94,33 +94,35 @@ class OpenocdError(ProcessError):
 class OpenocdGdbServer:
     """Handler for OpenocdGdbServer process"""
 
-    def __init__(self, interface: str, target: str):
+    def __init__(
+        self,
+        interface: Optional[str] = None,
+        target: Optional[str] = None,
+        board: Optional[str] = None,
+        extra_args: Optional[List[str]] = None,
+    ):
         self.proc = None
         self.target = target
         self.interface = interface
+        self.board = board
```

Comment:
right now board  and target/inferface are mutually exclusive (and changed to optional in typing), we might want to add checks for that, eg:

```python
if board:
    if target or interface:
        raise OpenocdError("target or interface arguments provided when board is specified")
else:
   if not target or not interface:
        raise OpenocdError("target or interface arguments missing")
```

probably adding docstring regarding this new semantics would also be helpful